fleetctl/fleetctl.go

```diff
@@ -1263,3 +1222,26 @@ func runWrapper(cf func(cCmd *cobra.Command, args []string) (exit int)) func(cCm
 		cmdExitCode = cf(cCmd, args)
 	}
 }
+
+// waitForState is a generic helper for repeatedly checking the status.
+// It gets a generic function stateCheckFunc() to be checked, which returns
+// true on success, false on error. In case of failure, waitForState
+// retries to run stateCheckFunc, once in 250 msec, up to defaultSleepTime.
+func waitForState(stateCheckFunc func() (bool, error)) (time.Duration, error) {
+	timeout := defaultSleepTime
+	alarm := time.After(timeout)
+	ticker := time.Tick(250 * time.Millisecond)
+
+	for {
+		select {
+		case <-alarm:
+			return timeout, fmt.Errorf("Failed to fetch systemd active states within %v", timeout)
+		case <-ticker:
+			isChecked, err := stateCheckFunc()
```

Comment:
Sorry to keep nitting on this, but I might not have explained clearly: `err` will _always_ be `nil` when `isChecked` is `true`, right? which means that you can just remove the `bool` value and rely on the nil-ness of the error :-)